x/ccv/provider/types/msg.go

```diff
@@ -746,112 +763,133 @@ func ValidateField(nameOfTheField string, field string, maxLength int) error {
+	if err := ValidateConsAddressList(powerShapingParameters.Allowlist, MaxValidatorCount); err != nil {
+		return errorsmod.Wrapf(ErrInvalidPowerShapingParameters, "Allowlist: %s", err.Error())
 	}
-
-	if len(powerShapingParameters.Denylist) > maxDenylistLength {
-		return fmt.Errorf("denylist cannot exceed length: %d", maxDenylistLength)
+	if err := ValidateConsAddressList(powerShapingParameters.Denylist, MaxValidatorCount); err != nil {
+		return errorsmod.Wrapf(ErrInvalidPowerShapingParameters, "Denylist: %s", err.Error())
 	}
 
 	return nil
 }
 
 // ValidateInitializationParameters validates that all the provided parameters are in the expected range
 func ValidateInitializationParameters(initializationParameters ConsumerInitializationParameters) error {
-	const maxGenesisHashLength = 1000
-	const maxBinaryHashLength = 1000
-	const maxConsumerRedistributionFractionLength = 50
-	const maxDistributionTransmissionChannelLength = 1000
-
 	if initializationParameters.InitialHeight.IsZero() {
-		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "initial height cannot be zero")
+		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "InitialHeight cannot be zero")
 	}
 
-	if len(initializationParameters.GenesisHash) == 0 {
-		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "genesis hash cannot be empty")
-	} else if len(initializationParameters.GenesisHash) > maxGenesisHashLength {
-		return errorsmod.Wrapf(ErrInvalidConsumerInitializationParameters, "genesis hash cannot exceed %d bytes", maxGenesisHashLength)
+	if err := validateHash(initializationParameters.GenesisHash); err != nil {
+		return errorsmod.Wrapf(ErrInvalidConsumerInitializationParameters, "GenesisHash: %s", err.Error())
 	}
 
-	if len(initializationParameters.BinaryHash) == 0 {
-		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "binary hash cannot be empty")
-	} else if len(initializationParameters.BinaryHash) > maxBinaryHashLength {
-		return errorsmod.Wrapf(ErrInvalidConsumerInitializationParameters, "binary hash cannot exceed %d bytes", maxBinaryHashLength)
+	if err := validateHash(initializationParameters.BinaryHash); err != nil {
+		return errorsmod.Wrapf(ErrInvalidConsumerInitializationParameters, "BinaryHash: %s", err.Error())
 	}
 
 	if initializationParameters.SpawnTime.IsZero() {
-		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "spawn time cannot be zero")
+		return errorsmod.Wrap(ErrInvalidConsumerInitializationParameters, "SpawnTime cannot be zero")
 	}
 
 	if err := ccvtypes.ValidateStringFraction(initializationParameters.ConsumerRedistributionFraction); err != nil {
-		return errorsmod.Wrapf(ErrInvalidConsumerInitializationParameters, "consumer redistribution fraction is invalid: %s", err.Error())
-	} else if err := ValidateField("consumer redistribution fraction", initializationParameters.ConsumerRedistributionFraction, maxConsumerRedistributionFractionLength); err != nil {
```

Comment:
I guess you removed the length check here because `ValidateStringFraction` has a `LegacyPrecision` limit?